src/state/timer-state.js

```diff
@@ -30,6 +32,10 @@ class TimerState {
         this.rotate()
         this.callback('turnEnded')
         this.startAlerts()
+
+        if (this.clearClipboardHistoryBetweenTurns) {
+          setTimeout(this.clearClipboardHistory, this.secondsUntilFullscreen * 1000)

```

Comment:
It looks like `clearClipboardHistory` will only ever happen once because of the use of `setTimeout` which only fires once.

Since you've named it using terminology of "BetweenTurns", then maybe the call to `clearClipboardHistory()` should be added into `rotate()`? If you do that, then there's no reason to have anything here in `createTimers()`.

But the downside of that is that then the clipboard history won't be cleared if you pause the timer; dunno if that fits with your concept of when/how often the clipboard should be cleared. It would probably help if you provided more context on what you want to accomplish with this change.

Reply:
This line is only intended to be called once, it is in the method `clearClipboardHistory` that I use `setInterval` to do an action multiple times.

My goal was to trigger clearing the clipboard history at the same time the full screen pops up with the message to rotate people. The only problem I see with adding it to `rotate()` is that `rotate()` gets called at other times such as when you skip a person.

Reply:
The `createTimers()` method is only ever called once and that's during application startup. So as your code is currently written, the timeout will be set when the mob-timer starts up and will execute only once 30 seconds after that (it will always be 30 seconds after, because this code will run before the configuration is loaded from disk and 30 seconds is the default value).

Clearly that's not what you want; you'd have to quit and restart the timer every time you want to clear the clipboard.

I do see that you used `setInterval` to control clearing the clipboard (I guess it doesn't work if you call it too rapidly in a loop?), but that's the internal implementation detail of how to clear the clipboard. I'm concerned that the overall functionality won't work as you expect.

The other problem you'll encounter is that if you tie clearing the clipboard to the fullscreen window, what happens when that is turned off?

The more I think about it, "between turns" is very vague. When does that happen exactly? Is it when a turn ends? When the next turn starts? Some time in between? And if in between, can it get skipped if the next mobber is really quick to start their turn?

Reply:
Maybe I'm misunderstanding what this section of code is doing - I thought this was creating the callback for the `mainTimer` and that this if blocked gets executed each time a person's turn ends.

Reply:
Derp. I was totally off base. You're right -- I glossed over the callback. So you're right that it will execute each time the `secondsRemaining < 0`. Sorry for leading you astray on that point.